src/lib/Libdb/pgsql/db_common.c

```diff
@@ -454,20 +457,11 @@ pbs_dataservice_control(char *cmd, char *pbs_ds_host, int pbs_ds_port)
 			oom_val = strdup("-17");
 		}
 		if (oom_val != NULL) {
-			if ((fd = open(oom_file, O_TRUNC | O_WRONLY, 0600)) == -1) {
-				if (errmsg_cache)
-					free(errmsg_cache);
-				errmsg_cache = strdup("OOM protect: file open failed");
-				return -1;
-			}
-			if (write(fd, oom_val, strlen(oom_val)) == -1) {
-				if (errmsg_cache)
-					free(errmsg_cache);
-				errmsg_cache = strdup("OOM protect: file write failed");
-				return -1;
+			if ((fd = open(oom_file, O_TRUNC | O_WRONLY, 0600)) != -1) {
+				(void)write(fd, oom_val, strlen(oom_val));
+				close(fd);
```

Comment:
I think we should still write the error message so that if OOM killer does kill it, we can see from logs that the OOM protect had failed

Reply:
I agree. We should have access to the logger from here, right?

Reply:
This is part of libdb which doesn't write to logs directly. The only way to pass the logs is through errmsg but if the next operation is a success we may not capture it.

Reply:
Ah, that sucks. In the caller, can we check if errmsg_cache is not null and log?

Reply:
No, not directly. We call pbs_db_get_errmsg() to get errmsg_cache in case of non-zero return code. I will have to introduce a new error code for OOM failure and make callers just log but still consider it as a successful start DB.

Reply:
as we discussed offline, if it's not a lot of effort to add a new return code, can you please try to do it?